elements/pfe-sass/functions/_custom-properties.scss

```
@@ -0,0 +1,134 @@
+/// Color
+///
+/// Returns CSS Var stack with exposed theme Var and respective SCSS fallback
+///
+/// @param {String} $cssvar - Variable name to be used and prepended with --pfe-theme
+/// @param {String} $category - Category name to be appended to variables within the map/system, default is "color"
+/// @param {Map} $map - SASS Map of variables
+/// @requires {String} $repo - Name of repo, which is "pfe"
+/// @use -
+///
+///     1.look up a color: /elements/pfe-sass/src/variables/_maps.scss
+///         (FYI, maps references: /elements/pfe-sass/src/variables/_colors.scss)
+///
+///     2. use like this:
+///         ```
+///         .my-element {
+///           color:       #{pfe-color(  surface--base--text  )};
+///           background:  #{pfe-color(  surface--base )};
+///         }
+///         ```
+
+@function pfe-color($cssvar, $category: 'color', $map: $pfe-colors) {
+    @if $category !="" {
+        $category: "#{$category}--";
+    }
+    // Start building the variable declaration
+    $var-declaration: "--#{$repo}-theme--#{$category}#{$cssvar}";
+    // If a fallback is defined, pass that in too
+    @if map-get($map, $cssvar) != null {
+        $var-declaration: "#{$var-declaration}, #{map-get($map, $cssvar)}";
+    }
+    // Return the variable declaration string
+    @return var(#{$var-declaration});
+}
+
+@function pfe-fallback($cssvar, $category: 'color', $map: $pfe-colors) {
+    @if $category !="" {
+        $category: "#{$category}--";
+    }
+    // Return the variable declaration string
+    @return #{map-get($map, $cssvar)};
+}
+
+/// Var
+///
+/// Returns CSS Var stack with exposed theme Var and respective SCSS fallback
+///
+/// @param {String} $cssvar - Variable name to be used and prepended with --pfe-theme
+/// @param {String} $category - Category name to be appended to variables within the map/system
+/// @param {Map} $map - SASS Map of variables
+/// @requires {String} $repo - Name of repo, which is "pfe"
+/// @use -
+///
+///     1.look up a color: /elements/pfe-sass/src/variables/_maps.scss
+///         (FYI, maps references: /elements/pfe-sass/src/variables/_colors.scss)
+///
+///     2. use like this:
+///         ```
+///         .my-element {
+///           padding:     #{pfe-var(  container-spacer  )};
+///           font-size:   #{pfe-var(  font-size  )};
+///         }
+///         ```
```

```suggestion
///         ```
///     3. which will return
///         ```
///         .my-element {
///            padding: var(--pfe-theme--container-spacer, 16px);
///            font-size: var(--pfe-theme--font-size, 16px);
///         }
///         ```
```